### PR TITLE
refactor: incorrect exception usage

### PR DESCRIPTION
**Pull request by bsherifi**
**Fixing Issue 2123:** 

Added 4 new type of errors (exceptions) in the _excepts_ module.

Refactored some of the modules in the code-base by replacing the _NotImplementedError_ with more suitable exceptions where it was needed.

### CODE REVIEW

**Review by Yongxuanzhang — COMMENTED** [review]
Hi @bsherifi, thanks for your contribution!
To pass the docstring and black CI, you may need to merge/rebase from the master, then follow this https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md
Thanks!
image

**Comment by bsherifi** [comment]
> Hi @bsherifi, thanks for your contribution!
> To pass the docstring and black CI, you may need to merge/rebase from the master, then follow this https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md
> Thanks!
> image

Thank you @Yongxuanzhang, now the docstring and black CI have passed. I'm having an issue with the docker test now, although it's not 'Required', it's making success-all-test fail. Let me know if I can do something about it.

**Comment by Yongxuanzhang** [comment]
> > Hi @bsherifi, thanks for your contribution!
> > To pass the docstring and black CI, you may need to merge/rebase from the master, then follow this https://github.com/jina-ai/jina/blob/master/CONTRIBUTING.md
> > Thanks!
> > image
> 
> Thank you @Yongxuanzhang, now the docstring and black CI have passed. I'm having an issue with the docker test now, although it's not 'Required', it's making success-all-test fail. Let me know if I can do something about it.

Just re-run the tests.

File: jina/peapods/pods/helper.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def _set_peas_args(
             elif args.scheduling == SchedulerType.LOAD_BALANCE:
                 _args.socket_in = SocketType.DEALER_CONNECT
             else:
-                raise NotImplementedError
+                raise ValueError
```
Comment (Yongxuanzhang) [inline jina/peapods/pods/helper.py:30]: Add error message here maybe?

File: jina/executors/indexers/vector.py [hunk 2]
```diff
@@ -495,7 +496,7 @@ def query(
             _query_vectors = _ext_A(_norm(vectors))
             dist = self._cosine(_query_vectors, self.query_handler)
         else:
-            raise NotImplementedError(f'{self.metric} is not implemented')
+            raise MetricTypeError(f'{self.metric} is not implemented')
```
Comment (Yongxuanzhang) [inline jina/executors/indexers/vector.py:499]: This line actually cannot be reached I think. Maybe need to refactor this part.

Comment (bsherifi) [reply jina/executors/indexers/vector.py]: Done

File: jina/peapods/pods/helper.py (hunk 1 above)
Comment (bsherifi) [reply jina/peapods/pods/helper.py]: Done

File: jina/peapods/peas/helper.py [hunk 3]
```diff
@@ -11,7 +11,9 @@ def _get_event(obj) -> Event:
     elif isinstance(obj, multiprocessing.Process):
         return multiprocessing.Event()
     else:
-        raise NotImplementedError
+        raise ValueError(
```
Comment (bwanglzu) [inline jina/peapods/peas/helper.py:14]:
```suggestion
        raise TypeError(
```

**Review by bwanglzu — COMMENTED** [review]
great work @bsherifi ! I noticed in some cases you created a customised exception, sometimes not, it might bring a bit inconsistency.

**Comment by bwanglzu** [comment]
mark it ready to review when you finish :)

**Comment by bsherifi** [comment]
> great work @bsherifi ! I noticed in some cases you created a customised exception, sometimes not, it might bring a bit inconsistency.

Thanks @bwanglzu! I created customized exceptions for the cases that seemed more special to me and probably needed a custom exception. I also figured the custom exceptions might be re-used in cases where it suits the description of it. However, I can turn that around and make all customized ones if you still think it would be better to do so.

**Comment by nan-wang** [comment]
recheckcla

**Comment by bsherifi** [comment]
> **Jina CLA check**
> 
>  Thank you for your pull request. It looks like this is your first contribution to an open source project maintained by Jina AI Limited. Before we can look at your pull request, we kindly ask that you sign our [Contributor License Agreement](https://github.com/jina-ai/jina/blob/master/.github/CLA.md). You can sign it by commenting in format below.
> 
> _**I have read the CLA Document and I hereby sign the CLA**_

I have read the CLA Document and I hereby sign the CLA

**Review by nan-wang — CHANGES_REQUESTED** [review]
Well done! Many thanks for your contribution! Minor change is wanted.

File: jina/peapods/pods/helper.py [hunk 4]
```diff
@@ -42,7 +42,10 @@ def _set_peas_args(
             elif args.scheduling == SchedulerType.LOAD_BALANCE:
                 _args.socket_in = SocketType.DEALER_CONNECT
             else:
-                raise NotImplementedError
+                raise ValueError(
+                    f"{args.scheduling} is not supported as a SchedulerType!"
```
[the remark this answers is not in the document]
Comment (bsherifi) [reply jina/peapods/pods/helper.py]: Thanks @nan-wang. I changed it to single quoted.

**Review by JoanFM — APPROVED** [review]
LGTM!

**Review by Yongxuanzhang — APPROVED** [review]

**Pull request merged**